Sources/Basics/FileSystem+Extensions.swift

```diff
@@ -69,52 +68,126 @@ extension FileSystem {
     }
 }
 
-// MARK: - config
+// MARK: - configuration
 
 extension FileSystem {
-    private var idiomaticUserConfigDirectory: AbsolutePath? {
-        return FileManager.default.urls(for: .libraryDirectory, in: .userDomainMask).first.flatMap { AbsolutePath($0.path) }
-    }
-
     /// SwiftPM config directory under user's config directory (if exists)
-    public var swiftPMConfigDirectory: AbsolutePath {
-        if let path = self.idiomaticUserConfigDirectory {
-            return path.appending(component: "org.swift.swiftpm")
+    public var swiftPMConfigurationDirectory: AbsolutePath {
+        if let path = self.idiomaticSwiftPMDirectory {
+            return path.appending(component: "configuration")
         } else {
-            return self.dotSwiftPMConfigDirectory
+            return self.dotSwiftPMConfigurationDirectory
         }
     }
 
-    fileprivate var dotSwiftPMConfigDirectory: AbsolutePath {
-        return self.dotSwiftPM.appending(component: "config")
+    fileprivate var dotSwiftPMConfigurationDirectory: AbsolutePath {
+        return self.dotSwiftPM.appending(component: "configuration")
+    }
+}
+
+extension FileSystem {
+    public func getOrCreateSwiftPMConfigurationDirectory() throws -> AbsolutePath {
+        let idiomaticConfigurationDirectory = self.swiftPMConfigurationDirectory
+
+        // temporary 5.6, remove on next version: transition from previous configuration location
+        do {
+            if !self.exists(idiomaticConfigurationDirectory) {
+                try self.createDirectory(idiomaticConfigurationDirectory, recursive: true)
+            }
+
+            if idiomaticConfigurationDirectory != self.dotSwiftPMConfigurationDirectory {
+                // move content from old directory to new one but create symlink for the files for backwards compatibility (eg older xcode)
+                let oldConfigDirectory = idiomaticConfigurationDirectory.parentDirectory
```

Comment:
Note to self: this moves all files under `~/Library/org.swift.swiftpm/` (e.g., `collections.json`) to `~/Library/org.swift.swiftpm/configuration/`.